src/Saver/UploadSaver.php

```diff
@@ -10,15 +10,18 @@ final class UploadSaver implements SaverInterface
     private $url;
     /** @var int */
     private $timeout;
+    /** @var bool */
+    private $verify;
 
-    public function __construct($url, $token, $timeout)
+    public function __construct($url, $token, $timeout, $verify)
     {
         $this->url = $url;
         if ($token) {
             $this->url .= '?&token=' . $token;
         }
 
         $this->timeout = $timeout;
+        $this->verify = is_bool($verify) ? $verify : true;
```

Comment:
```suggestion
        $this->verify = $verify;
```

superfluous check: default is initialized in factory class, if they pass garbage, instead of value of proper type, they know what are they doing. 

also other parameters are not checked, so this hidden validation could rather be unexpected